HIP/hip-968.md

```diff
@@ -0,0 +1,131 @@
+---
+hip: 968
+title: Enhance mirror node rest API to query by token name
+author: Simi Hunjan <SimiHunjan>, Ali Nikan <@alinik4n>
+working-group: Steven Sheehy <@steven-sheehy> Eric <ericleponner> Simon <svienot>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Review
+created: 2024-04-19
+updated: 2024-05-14
+requested-by: Hashport, Bonzo Finance
+---
+
+## Abstract
+
+This HIP provides the capability of querying tokens by token name in mirror node REST API and getting the corresponding token ID information.
+
+## Motivation
+
+To enhance user experience and allow blockchain explorers such as Hashscan to enable the capability of searching by token name, we want to add this functionality to the mirror node to support querying tokens by token name.
+This is a very common use case across other explorers in the larger ecosystem, and will allow retail users and developers to be able to search for a token by token name instead of token ID.
```

Comment:
Q: On other chains the API is on the explorer I believe, at least the EVM ones. 
In this case we're adding it to the mirror node which has impacts on requests and db perf since token name isn't a unique identifier. 
Was their a design choice here for mirror node vs mirror node explorer?

If there was it could be added to rejected ideas as something previously explored

Reply:
The mirror node explorer does not store any data or have its own queries for which it returns information for. It was one of the design choices made early on when the explorer was being developed from what I've been told and since then this has not changed. As @ericleponner would say mirror node explorer is the client part of mirror node.  So when we search to return something in the explorer it is mapped to a specific mirror node query. We can consider things to be in the explorer in a future state when we have support to do these kinds of things.